lib/routes/key_data.js

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Joi = require('joi');
+
+const AppError = require('../error');
+const db = require('../db');
+const logger = require('../logging')('routes.key_data');
+const P = require('../promise');
+const validators = require('../validators');
+const verify = require('../browserid');
+const Scope = require('../scope');
+
+const ADDITIONAL_KEY_MATERIAL = Buffer.alloc(32).toString('hex');
+
+module.exports = {
+  validate: {
+    params: {
+      client_id: validators.clientId
+    },
+    payload: {
+      assertion: validators.assertion.required(),
+      scope: Joi.string()
+    }
+  },
+  response: {
+    schema: Joi.object().pattern(/^/, [
+      Joi.object({
+        identifier: Joi.string().required(),
+        keyMaterial: Joi.string().required(),
+        timestamp: Joi.number().required()
+      })
+    ])
+  },
+  handler: function keyDataRoute(req, reply) {
+    logger.debug('keyDataRoute.start', {
+      params: req.params,
+      payload: req.payload
+    });
+
+    const requestedScopes = new Set(Scope(req.payload.scope).values());
```

Comment:
I don't think you need to turn this into a `Set`; the `Scope` class contains `has` method that IIUC should do the right thing?